core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/BaseAwsJsonProtocolFactory.java

```diff
@@ -322,10 +319,40 @@ public final SubclassT hasAwsQueryCompatible(boolean hasAwsQueryCompatible) {
             return getSubclass();
         }
 
+        /**
+         * Provides the unmarshalling instant registry factory to be used to unmarshall {@link Instant} values in requests.
+         * This is only expected to be used by extending protocols, such as RPCv2, that requires a different registry to be
+         * able to read embedded objects instead of converting them to string and then back into a value.
+         *
+         * @param timestampFormatRegistryFactory the factory to create an unmarshall registry
+         * @return This builder for method chaining.
+         */
+        protected final SubclassT timestampFormatRegistryFactory(
+            TimestampFormatRegistryFactory timestampFormatRegistryFactory
+        ) {
+            this.timestampFormatRegistryFactory = timestampFormatRegistryFactory;
+            return getSubclass();
+        }
```

Comment:
Is there a way to make this not specific to timestamps? It feels better to have protocol-specific marshalling for any type, not just Instants.

Reply:
Yeah, that would be ideal, not sure how though, suggestions?. I'm trying to generalize the functionality statically defined before in the [`createUnmarshallerRegistry`](https://github.com/aws/aws-sdk-java-v2/blob/3bc3427813e75a86530714075dcc67d37e1e9acd/core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/internal/unmarshall/JsonProtocolUnmarshaller.java#L72)

Reply:
Oh, the formats themselves are part of the protocol, so yeah, let me make that happen.

Reply:
Pushed a commit to change this, now we take a single object with all the dependencies which include the registry.